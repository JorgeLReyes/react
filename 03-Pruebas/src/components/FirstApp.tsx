import PropTypes from "prop-types";

interface Props {
  title: string;
  subTitle: string;
  name: string;
}

const FirstApp = ({ title, subTitle, name }: Props) => {
  return (
    <div>
      <h1 data-testid="test-title">{title}</h1>
      <h2>{subTitle}</h2>
      <h2>{subTitle}</h2>
      <p>Hola, {name}!</p>
    </div>
  );
};

// FirstApp.defaultProps = {
//   title: "Hello World",
//   subTitle: "Welcome to my app",
//   // name: "Ingrid",
// };

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};

export default FirstApp;
