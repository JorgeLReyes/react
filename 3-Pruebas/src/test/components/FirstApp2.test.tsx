import { render, screen } from "@testing-library/react";
import FirstApp from "../../components/FirstApp";
describe("Component FirstApp 2", () => {
  const options = {
    title: "Hi",
    subTitle: "World",
    name: "Ingi",
  };

  test("should be match with snapshot", () => {
    const { container } = render(<FirstApp {...options} />);
    expect(container).toMatchSnapshot();
  });

  test("should be see message 'Mundo'", () => {
    render(<FirstApp {...options} />);
    // screen.debug();
    const { debug, getByText } = screen;
    debug();
    expect(getByText(options.title)).toBeTruthy();
  });

  test("should be watch title in h1", async () => {
    render(<FirstApp {...options} />);
    expect(
      screen.getByRole("heading", { level: 1, name: options.title })
    ).toBeTruthy();
  });

  test("should be watch subtitle by props", async () => {
    render(<FirstApp {...options} />);

    expect(
      screen.getAllByRole("heading", { level: 2, name: options.subTitle })
        .length
    ).toBe(2);
  });
});
