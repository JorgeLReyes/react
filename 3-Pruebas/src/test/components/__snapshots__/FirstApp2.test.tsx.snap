// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component FirstApp 2 should be match with snapshot 1`] = `
<div>
  <div>
    <h1
      data-testid="test-title"
    >
      Hi
    </h1>
    <h2>
      World
    </h2>
    <h2>
      World
    </h2>
    <p>
      Hola, 
      Ingi
      !
    </p>
  </div>
</div>
`;
